core/flash/spi_flash.c

```diff
@@ -192,7 +192,7 @@ static int spi_flash_configure_device (const struct spi_flash *flash, bool wake_
 
 	if (reset_device) {
 		status = spi_flash_reset_device (flash);
-		if (status != 0) {
+		if ((status != 0) && (status != SPI_FLASH_RESET_NOT_SUPPORTED)) {
```

Comment:
This is an interesting scenario.  On the one hand, if you are saying you want the flash device reset, that request has failed if the device doesn't support the operation.  On the other hand, I could see the use-case where you want the reset to be a "best effort", meaning reset if you can.  I wonder if we need separate initialize_device APIs to cover each case.

Also, there need to be unit tests submitted with code changes.

Reply:
I think the best option is to split this long function to sub functions:
1. spi_flash_init : read_id + SFDP.
2. spi_flash_config: fast_read\quad\drive strength etc.
3. spi_flash_reset: only reset.

otherwise for a flash that doesn't support reset the function will not do all the steps that happen after the reset.

API is a drastic change, so I tried to keep it minimal. 

I will try to add a new unit test for this and resubmit this patch. 

thanks, Chris!